packages/oscal-react-library/src/components/OSCALMetadataCommon.tsx

```diff
@@ -1,12 +1,405 @@
+import { Address, TelephoneNumber } from "@easydynamics/oscal-types";
+import BusinessIcon from "@mui/icons-material/Business";
+import InfoIcon from "@mui/icons-material/Info";
+import ErrorOutlineIcon from "@mui/icons-material/ErrorOutline";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import HelpOutlineIcon from "@mui/icons-material/HelpOutline";
+import HomeIcon from "@mui/icons-material/Home";
+import SmartphoneIcon from "@mui/icons-material/Smartphone";
+import {
+  Dialog,
+  CardActions,
+  Avatar,
+  CardHeader,
+  ListItem,
+  Stack,
+  Typography,
+  Button,
+} from "@mui/material";
+import Accordion from "@mui/material/Accordion";
+import AccordionDetails from "@mui/material/AccordionDetails";
+import AccordionSummary from "@mui/material/AccordionSummary";
+import Grid from "@mui/material/Grid";
+import Link from "@mui/material/Link";
 import { styled } from "@mui/material/styles";
-import { Typography } from "@mui/material";
+import { default as React, ReactNode, useEffect } from "react";
+import OSCALAnchorLinkHeader from "./OSCALAnchorLinkHeader";
+import { OSCALMarkupLine } from "./OSCALMarkupProse";
 
 /**
  * A collection of common components shared amongst
  * OSCALMetadata child components.
  */
 
+const OSCALMetadataCardTitleFallbackText = styled(Typography)(({ theme }) => ({
+  color: theme.palette.text.secondary,
+})) as typeof Typography;
+
 export const OSCALMetadataLabel = styled(Typography)(({ theme }) => ({
   textAlign: "right",
   color: theme.palette.text.secondary,
 })) as typeof Typography;
+
+const OSCALMetadataSectionInfoHeader = styled(Typography)`
+  display: flex;
+  align-items: center;
+` as typeof Typography;
+
+interface MetadataAvatarProps {
+  /**
+   * String to create avatar color with
+   */
+  id: string;
+  /**
+   * Text to appear on avatar
+   */
+  text: string | undefined;
+  /**
+   * Icon to use if text is undefined
+   */
+  fallbackIcon: ReactNode;
```

Comment:
Can you see if there's some MUI type that we can use here for an icon type?

Reply:
I looked around and couldn't find anything too useful.